services/QuillLMS/app/models/subscription.rb

```diff
@@ -50,6 +50,10 @@ class RenewalNilStripeCustomer < StandardError; end
   after_commit :check_if_purchaser_email_is_in_database
   after_initialize :set_null_start_date_to_today
 
+  DEFAULT_SUMMER_CUTOFF_MONTH = 7
```

Comment:
Good call on extracting these constants out.

I made the mistake of not making the coupling to these constants explicit with [BonusDaysCalculator](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/models/bonus_days_calculator.rb#L4)

You'll need to add a couple constants to Subscription and then reference them in the BonusDaysCalculator:

```
class BonusDaysCalculator < ApplicationService
  SUMMER_DAY = Subscription::DEFAULT_SUMMER_CUTOFF_DAY
  SUMMER_MONTH = Subscription::DEFAULT_SUMMER_CUTOFF_MONTH

  WINTER_DAY = Subscription::DEFAULT_WINTER_CUTOFF_DAY
  WINTER_MONTH = Subscription::DEFAULT_WINTER_CUTOFF_MONTH

  ...

  private def plan_ends
    start.month <= SUMMER_MONTH ? start.change(month: SUMMER_MONTH , day: SUMMER_DAY) : start.change(month: WINTER_MONTH, day: WINTER_DAY)
  end
```

Reply:
Got it, I've changed those constants!